### PR TITLE
Add doc examples for std::alloc::{alloc,alloc_zeroed}.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks!

r=me once CI is ok.

File: src/liballoc/alloc.rs [hunk 1]
```diff
@@ -56,6 +56,21 @@ pub struct Global;
...
 ///
 /// See [`GlobalAlloc::alloc`].
+///
+/// # Examples
+///
+/// ```
+/// use std::alloc::{alloc, dealloc, Layout};
+///
+/// unsafe {
+///     let layout = Layout::from_size_align(2, 1).unwrap();
```
Comment (ollie27) [inline src/liballoc/alloc.rs]: `u16` generally has an alignment of 2 not 1. Using `Layout::new::<u16>()` would be better here though.

File: src/liballoc/alloc.rs [hunk 2]
```diff
@@ -110,6 +125,21 @@ pub unsafe fn realloc(ptr: *mut u8, layout: Layout, new_size: usize) -> *mut u8
...
 ///
 /// See [`GlobalAlloc::alloc_zeroed`].
+///
+/// # Examples
+///
+/// ```
+/// use std::alloc::{alloc_zeroed, dealloc, Layout};
+///
+/// unsafe {
+///     let layout = Layout::from_size_align(2, 1).unwrap();
```
Comment (ollie27) [inline src/liballoc/alloc.rs]: Again, this should be `Layout::new::<u16>()`.

File: src/liballoc/alloc.rs [hunk 3]
```diff
@@ -56,6 +56,21 @@ pub struct Global;
...
+/// # Examples
+///
+/// ```
+/// use std::alloc::{alloc, dealloc, Layout};
+///
+/// unsafe {
+///     let layout = Layout::from_size_align(2, 1).unwrap();
+///     let ptr = alloc(layout);
+///
+///     println!("Pointer {:p} has data: {:b}", ptr, *(ptr as *mut u16));
```
Comment (ollie27) [inline src/liballoc/alloc.rs]: This is reading uninitialised memory which is Undefined Behaviour. I suggest writing a value to it first and then reading that value back.

**Review by ollie27 — CHANGES_REQUESTED** [review]

**Comment by frewsxcv** [comment]
@ollie27 Thanks for the feedback! Comments addressed in the latest force push

**Comment by ollie27** [comment]
Looks good now.

@bors r+ rollup

**Pull request merged**